modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItSqlAsynchronousApiTest.java

```diff
@@ -0,0 +1,434 @@
+                ses,
+                "ALTER TABLE NOT_EXISTS_TABLE DROP COLUMN VAL1"
+        );
+        checkDdl(false, ses, "ALTER TABLE IF EXISTS NOT_EXISTS_TABLE DROP COLUMN VAL1");
+        checkError(
+                ColumnNotFoundException.class,
+                "Column 'VAL1' does not exist in table '\"PUBLIC\".\"TEST\"'",
+                ses,
+                "ALTER TABLE TEST DROP COLUMN VAL1"
+        );
+        checkDdl(false, ses, "ALTER TABLE TEST DROP COLUMN IF EXISTS VAL1");
+
+        // DROP TABLE
+        checkDdl(false, ses, "DROP TABLE IF EXISTS NOT_EXISTS_TABLE");
+        checkDdl(true, ses, "DROP TABLE TEST");
+        checkError(
+                TableNotFoundException.class,
+                "Table does not exist [name=PUBLIC.TEST]",
+                ses,
+                "DROP TABLE TEST"
+        );
+    }
+
+    @Test
+    public void dml() throws ExecutionException, InterruptedException {
+        sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
+
+        IgniteSql sql = CLUSTER_NODES.get(0).sql();
+        Session ses = sql.createSession();
+
+        for (int i = 0; i < ROW_COUNT; ++i) {
+            checkDml(1, ses, "INSERT INTO TEST VALUES (?, ?)", i, i);
+        }
+
+        checkDml(ROW_COUNT, ses, "UPDATE TEST SET VAL0 = VAL0 + ?", 1);
+
+        checkDml(ROW_COUNT, ses, "DELETE FROM TEST WHERE VAL0 >= 0");
+    }
+
+    @Test
+    public void select() throws ExecutionException, InterruptedException {
+        sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
+        for (int i = 0; i < ROW_COUNT; ++i) {
+            sql("INSERT INTO TEST VALUES (?, ?)", i, i);
+        }
+
+        IgniteSql sql = CLUSTER_NODES.get(0).sql();
+        Session ses = sql.sessionBuilder().defaultPageSize(ROW_COUNT / 4).build();
+
+        TestPageProcessor pageProc = new TestPageProcessor(4);
+        ses.executeAsync(null, "SELECT ID FROM TEST").thenCompose(pageProc).get();
+
+        Set<Integer> rs = pageProc.result().stream().map(r -> r.intValue(0)).collect(Collectors.toSet());
+
+        for (int i = 0; i < ROW_COUNT; ++i) {
+            assertTrue(rs.remove(i), "Results invalid: " + pageProc.result());
+        }
+
+        assertTrue(rs.isEmpty());
+    }
+
+    @Test
+    public void sqlRow() throws ExecutionException, InterruptedException {
+        IgniteSql sql = CLUSTER_NODES.get(0).sql();
+        Session ses = sql.sessionBuilder().build();
+
+        AsyncResultSet ars = ses.executeAsync(null, "SELECT 1 as COL_A, 2 as COL_B").get();
+
+        SqlRow r = CollectionUtils.first(ars.currentPage());
+
+        assertEquals(2, r.columnCount());
+        assertEquals(0, r.columnIndex("COL_A"));
+        assertEquals(1, r.columnIndex("COL_B"));
+        assertEquals(-1, r.columnIndex("notExistColumn"));
+
+        assertEquals(1, r.intValue("COL_A"));
+        assertEquals(2, r.intValue("COL_B"));
+
+        assertThrowsWithCause(
```

Comment:
let's add one more assertion to validate IndexOutOfBound is thrown when trying to access column by index which is not exists